main.go

```diff
@@ -155,6 +156,10 @@ func do() (err error) {
 		}
 	}
 
+	if len(gopts.SingleFile) > 0 && filepath.Ext(gopts.SingleFile) != ".go" {
+		return fmt.Errorf("Name: %v invalid. A {FILENAME}.go name must be provided", gopts.SingleFile)
```

Comment:
name %q is invalid

Error messages must begin with lower case (https://github.com/golang/go/wiki/CodeReviewComments#error-strings) and you want to use `%q` to have it quote things like empty strings.